crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -50,4 +64,1082 @@ impl PackageInference {
+        let mut walked_dependents = HashSet::new();
+        let mut walked_dependent_dependencies = HashSet::new();
+        let mut cherry_picked_packages = HashSet::new();
+
+        for selector in selectors {
+            let selector_packages = self.filter_graph_with_selector(&selector)?;
+            let no_selectors = selector_packages.is_empty();
+
+            for package in selector_packages {
+                let node = package_graph::WorkspaceNode::Workspace(
+                    package_graph::WorkspaceName::Other(package.clone()),
+                );
+
+                if selector.include_dependencies {
+                    let dependencies = self.pkg_graph.dependencies(&node);
+                    let dependencies = dependencies
+                        .iter()
+                        .flat_map(|i| i.name().map(str::to_string))
+                        .collect::<Vec<_>>();
+
+                    // flatmap through the option, the set, and then the optional package name
+                    walked_dependencies.extend(dependencies);
+                }
+
+                if selector.include_dependents {
+                    let dependents = self.pkg_graph.dependents(&node);
+                    for dependent in dependents.iter().flat_map(|i| i.name().map(str::to_string)) {
+                        walked_dependents.insert(dependent.clone());
+
+                        // get the dependent's dependencies
+                        if selector.include_dependencies {
+                            let dependent_node = package_graph::WorkspaceNode::Workspace(
+                                package_graph::WorkspaceName::Other(dependent),
+                            );
+
+                            let dependent_dependencies =
+                                self.pkg_graph.dependencies(&dependent_node);
+
+                            let x = dependent_dependencies
+                                .iter()
+                                .flat_map(|i| i.name().map(str::to_string))
+                                .collect::<HashSet<_>>();
+
+                            walked_dependent_dependencies.extend(x);
+                        }
+                    }
+                }
+
+                if (selector.include_dependents || selector.include_dependencies)
+                    && !selector.exclude_self
+                {
+                    // if we are including dependents or dependencies, and we are not excluding
+                    // ourselves, then we should add ourselves to the list of packages
+                    walked_dependencies.insert(package);
+                } else if !selector.include_dependencies && !selector.include_dependents {
+                    // if we are neither including dependents or dependencies, then
+                    // add  to the list of cherry picked packages
+                    cherry_picked_packages.insert(package);
+                }
+            }
+
+            if no_selectors {
```

Comment:
Nit: I think `selector_packages.is_empty()` is actually more descriptive here than pulling it out into a local variable, especially since the definition of `no_selectors` is a little far away.

Reply:
I moved the case up and put in a continue